gix-worktree/src/index/status.rs

```diff
@@ -0,0 +1,60 @@
+//!
+
+use bstr::BStr;
+
+///
+pub mod worktree;
+
+///
+pub mod index;
+
+///
+pub mod recorder;
+
+///
+pub mod diff;
+
+/// The status of an index entry in a worktree
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Debug, Default)]
+pub enum Status<T = ()> {
+    #[default]
+    /// The file in the worktree is identical to the index entry
+    Unchanged,
+    /// An index entry has no corresponding file in the worktree.
+    Removed,
+    /// The type of file changed (symlink <=> file) performing a
+    /// diff is usually not necessary/desired
+    TypeChange,
+    /// The worktree file that belongs to this index has a changed stat
+    /// therefore could have been modified.
+    ///
+    /// Note that this doesn't necessarily mean that the *content* of the file changed.
+    /// A modified even is emitted whenever a change could have occurred. It is up
+    /// to API consumers to check the file for content changes
+    Modified {
+        /// Indicates that one of the stat changes was an executable bit change
+        /// which is a significant change itself (for git status)
+        /// these files don't need to be rechanged
+        executable_bit_changed: bool,
+        /// The output of the diff run on this entry.
+        /// if the there is no content change and only the executable bit
+        /// changed than this is `None`
+        diff: Option<T>,
```

Comment:
So the entire idea behind calling this `diff` (same for the trait) was that I actually didn't model this after the git status code, because there is no such thing. There is a `git diff` code (worktree to index) that is also used by `git status` (and every other command like merge, rebase, add,...).

I didn't do it in this PR but the idea was that `git diff` would basically create a `String` here with a unified diff which then gets printed.

Reply:
The new name `content_change` is definitely makes sense to me, thank you.